test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,430 @@
+
+        const char *key_prefix;
+        testutil_check(cursor_prefix->get_key(cursor_prefix.get(), &key_prefix));
+        std::string key_prefix_str = key_prefix;
+
+        /* The key at the prefix cursor should contain the prefix. */
+        testutil_assert(key_prefix_str.substr(0, prefix.size()) == prefix);
+
+        logger::log_msg(LOG_TRACE,
+          "search_near (normal) exact " + std::to_string(exact_default) + " key " + key_default);
+        logger::log_msg(LOG_TRACE,
+          "search_near (prefix) exact " + std::to_string(exact_prefix) + " key " + key_prefix);
+
+        /* Example: */
+        /* keys: a, bb, bba */
+        /* Default search_near(bb) returns a, exact = -1 */
+        /* Prefix search_near(bb) returns bb, exact = 0 */
+        /* Prefix search_near(bb) returns bba, exact = 1 */
+        if (exact_default == -1) {
+            /* The key at the default cursor should not contain the prefix. */
+            testutil_assert((key_default_str.substr(0, prefix.size()) != prefix));
+
+            /*
+             * At least all the keys between the default cursor and the prefix cursor should contain
+             * the prefix.
+             */
+            while (ret = cursor_default->next(cursor_default.get()) == 0) {
+                testutil_check(cursor_default->get_key(cursor_default.get(), &k));
+                k_str = k;
+                testutil_assert((k_str.substr(0, prefix.size()) == prefix));
+                if (k_str == key_prefix_str)
+                    break;
+            }
+
+            /* We should be at the key pointed by the prefix cursor. */
+            testutil_assert(ret != WT_NOTFOUND && k_str == key_prefix_str);
+        }
+        /* Example: */
+        /* keys: a, bb, bba */
+        /* Default search_near(bb) returns bb, exact = 0 */
+        /* Prefix search_near(bb) returns bb, exact = 0 */
+        /* Prefix search_near(bb) returns bba, exact = 1 */
+        else if (exact_default == 0) {
+            /* The key at the default cursor should be the prefix. */
+            testutil_assert(key_default_str == prefix);
+
+            if (exact_prefix == 0) {
+                /* Both cursors should be pointing at the same key. */
+                testutil_assert(key_default_str == key_prefix_str);
+            } else {
+                /*
+                 * The next key of the default cursor should be equal to the key pointed by the
+                 * prefix cursor.
+                 */
+                testutil_assert(cursor_default->next(cursor_default.get()) == 0);
+                testutil_check(cursor_default->get_key(cursor_default.get(), &k));
+                testutil_assert(k == key_prefix_str);
+            }
+        }
+        /* Example: */
+        /* keys: a, bb, bba */
+        /* Default search_near(bb) returns bba, exact = 1 */
+        /* Prefix search_near(bb) returns bb, exact = 0 */
+        /* Prefix search_near(bb) returns bba, exact = 1 */
```

Comment:
don't need to worry for returning "bba" case. Since bb should be always the value it returns.

Reply:
What if `bb` is not visible?

Reply:
I think in this case, it is not possible to have `exact_prefix` to be equal to 0 if `exact_prefix` is not 0.

Reply:
I agree. `exact_prefix` can't be 0 if `exact_default` is 0.

Reply:
Fixed in Fixed in a5602ced362d3cf2d7bf95b2e4b8469689f4265a